src/System.Runtime.Extensions/tests/System/IO/PathTests.cs

```diff
@@ -46,6 +45,15 @@ public static void GetDirectoryName(string path, string expected)
         }
 
         [Theory]
+        [InlineData("")]
+        [InlineData(" ")]
+        [InlineData("\0")]
+        public static void GetDirectoryName_InvalidPaths_Throws(string path)
+        {
+            AssertExtensions.Throws<ArgumentException>("path", null, () => Path.GetDirectoryName(path));
+        }
```

Comment:
Similarly, a test for this was added here:
https://github.com/dotnet/corefx/pull/19440/commits/e1755b5a62cd07037d5488d459a61cf97378226d#diff-19631815f66f1f0ad0492da989a7463cR38